core/src/main/java/com/google/googlejavaformat/java/Main.java

```diff
@@ -187,6 +189,19 @@ private int formatFiles(CommandLineOptions parameters, JavaFormatterOptions opti
         outWriter.write(formatted);
       }
     }
+
+    // #846 Clean up any resources and threads created by the executorService.
+    final boolean completedShutdown =
+        MoreExecutors.shutdownAndAwaitTermination(executorService, 100, TimeUnit.MILLISECONDS);
+
+    if (!completedShutdown) {
+      // We wait for all formatting tasks to complete when looping through the formatting tasks
+      // before shutdown, so no tasks should be left over by the time we shut down the executor
+      // service.
+      throw new IllegalStateException(
+          "Failed to complete all formatting tasks and shut down the formatter.");
```

Comment:
I think this should print the message to stderr and return with a non-zero exit code, similar to the error handling on other code paths here.